translator/util/sdkutil.go

```diff
@@ -38,6 +39,11 @@ func DetectAgentMode(configuredMode string) string {
 		return config.ModeEC2
 	}
 
+	if runWithIrsa == config.RUN_WITH_IRSA_TRUE {
```

Comment:
Would there be any reason why this won't be the last check? (e.g after `if DefaultECSRegion() != ""{)`)

Reply:
No. Why would this check get prioritized  over all other checks and not as a last check before return its as onPrem

Reply:
e.g Why not putting it here
```
if DefaultEC2Region() != "" {
		fmt.Println("I! Detected the instance is EC2")
		return config.ModeEC2
	}
	if DefaultECSRegion() != "" {
		fmt.Println("I! Detected the instance is ECS")
		return config.ModeEC2
	}
        if runWithIrsa == config.RUN_WITH_IRSA_TRUE {
               return config.ModeWithIRSA
        }
	fmt.Println("I! Detected the instance is OnPremise")
	return config.ModeOnPrem
```
but the reverse?

Reply:
Because the RUN_WITH_IRSA is an override flag. It should come before we try to ping IMDS and ECS

Reply:
Sounds fair to me !